pkg/controller/statusmanager/pod_status.go

```diff
@@ -247,13 +246,16 @@ func (status *StatusManager) setNodeConditionFromPod(podName types.NamespacedNam
 	// received the outdated pot event after the new pod running. For this case
 	// we should ignore the outdated event, otherwise, the operator may mistakenly
 	// set the NetworkUnavailbe status.
-	podStartedAt := pod.Status.StartTime.Time
-	nodeLastStartedAt := sharedInfo.LastNodeAgentStartTime[nodeName]
-	if podStartedAt.Before(nodeLastStartedAt) {
-		log.Info("Pod %s started at %v on node %s is outdated, there's new pod started at %v", podStartedAt, podName, nodeName, nodeLastStartedAt)
-		return
-	} else {
-		sharedInfo.LastNodeAgentStartTime[nodeName] = podStartedAt
+	// StartTime field will not be set until pod is initilized in pending state
+	if (pod.Status.StartTime != nil) {
```

Comment:
Do we need to return if `pod.Status.StartTime == nil`?

Reply:
I think it's ok we can continue when pod.Status.StartTime == nil.  
we have logic below in code line 262 to handle pending state before Status.StartTime is set. 

// when pod is pending, its ContainerStatuses is empty